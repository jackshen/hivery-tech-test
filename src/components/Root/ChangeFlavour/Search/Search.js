import React from "react";
import styled from "styled-components";

const SearchInput = styled.input.attrs({
  placeholder: "Search",
  type: "text"
})`
  border: 0.0625rem solid ${props => props.theme.veryLightGrey};
  border-radius: 0.25rem;
  box-sizing: border-box;
  color: ${props => props.theme.raven};
  font: inherit;
  height: 2.5625rem;
  outline: none;
  padding-left: 1.125rem;
  padding-right: 2.375rem;
  width: 17.0625rem;
`;

const SearchWrapper = styled.div`
  grid-area: search;
  height: fit-content;
  justify-self: end;
  margin: 2.215rem 0;
  position: relative;
  width: fit-content;

  :after {
    border: 0.125rem solid ${props => props.theme.nobel};
    border-radius: 0.125rem;
    box-sizing: border-box;
    color: ${props => props.theme.nobel};
    content: "?";
    display: inline-block;
    font-weight: bold;
    height: 1.375rem;
    position: absolute;
    right: 0.5rem;
    text-align: center;
    top: 50%;
    transform: translate(0, -50%);
    width: 1.375rem;
  }
`;

const Search = ({ onChange, value }) => (
  <SearchWrapper>
    <SearchInput onChange={onChange} value={value} />
  </SearchWrapper>
);

export default Search;
