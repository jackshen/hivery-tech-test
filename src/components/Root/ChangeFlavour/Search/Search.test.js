// including purely a snapshot test like this has its advantages and disadvantages
// the main advantage being that accidental changes to things such as the Styled Components' theme
// would not cause stylistic breaks

import React from "react";

import wrapAll from "#test/wrapAll";

import Search from "./Search";

describe("Search", () => {
  // we do not test value or onChange here - those are tested in integration tests found in ChangeFlavour
  test("matches snapshot", () => {
    const { container } = wrapAll(<Search />);

    expect(container).toMatchSnapshot();
  });
});
