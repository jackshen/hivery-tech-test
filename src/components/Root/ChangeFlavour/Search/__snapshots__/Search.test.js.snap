// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches snapshot 1`] = `
.c1 {
  border: 0.0625rem solid #cccccc;
  border-radius: 0.25rem;
  box-sizing: border-box;
  color: #6e7074;
  font: inherit;
  height: 2.5625rem;
  outline: none;
  padding-left: 1.125rem;
  padding-right: 2.375rem;
  width: 17.0625rem;
}

.c0 {
  grid-area: search;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  justify-self: end;
  margin: 2.215rem 0;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0:after {
  border: 0.125rem solid #999999;
  border-radius: 0.125rem;
  box-sizing: border-box;
  color: #999999;
  content: "?";
  display: inline-block;
  font-weight: bold;
  height: 1.375rem;
  position: absolute;
  right: 0.5rem;
  text-align: center;
  top: 50%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  width: 1.375rem;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
</div>
`;
